lib/charms/layer/apache_zeppelin.py

```diff
@@ -13,15 +24,37 @@ class Zeppelin(object):
 
     :param DistConfig dist_config: The configuration container object needed.
     """
+    @classmethod
+    def get(cls):
+        """
+        Dynamically instantiate this or a subclass, to allow other layers
+        to override portions of this implementation.
+        """
+        impl = layer.options('apache-zeppelin')['implementation_class']
+        module_name, cls_name = impl.rsplit('.', 1)
+        cls = getattr(import_module(module_name), cls_name)
+        return cls()
```

Comment:
If you're going the factory approach, then I'd just rename .get to .factory (as done here http://python-3-patterns-idioms-test.readthedocs.io/en/latest/Factory.html). It does the exact same thing, but I think that it flags what you're doing more clearly.